### PR TITLE
PUT endpoint for updating an existing post

### PR DESCRIPTION
**Pull request by maaxolofsson**
Tested in Insomnia with /posts/1 as endpoints, given the body:
```json
{
  "content": "Updated post"
}
```
and received:
```json
{
	"status": "success",
	"data": {
		"id": 1,
		"content": "Updated post",
		"createdAt": "2024-10-31T09:03:37.276Z",
		"updatedAt": "2024-10-31T09:45:43.651Z",
		"userId": 1,
		"user": {
			"id": 1,
			"role": "STUDENT",
			"email": "student@test.com",
			"profile": {
				"firstName": "Joe",
				"lastName": "Bloggs",
				"bio": "Hello, world!",
				"githubUrl": "student1"
			}
		}
	}
}
```

Also tested so that the error codes are returning correctly (post not found, bad request body, etc.) as defined in the API spec.

### CODE REVIEW

**Comment by LudwigJL** [comment]
Looks good!

**Review by LudwigJL — APPROVED** [review]
Looks good

**Review by sebgro98 — COMMENTED** [review]
Should anyone be able to edit anyones post? or will we have validation on the frontend regardig this.

I think best solution would be that you can only edit your own post, so we would check if the logged in user has the same id as the one being sent?

File: src/controllers/post.js [hunk 1]
```diff
@@ -40,3 +40,30 @@ export const getById = async (req, res) => {
...
+
+  if (!content) {
+    return sendDataResponse(res, 400, { content: 'Must provide content' })
+  }
+
+  const checkIfPostExists = await Post.findById(id)
+  if (!checkIfPostExists) {
+    return sendDataResponse(res, 404, { id: 'Post not found' })
+  }
+
```
Comment (maaxolofsson) [inline src/controllers/post.js:60]:
```JavaScript
const { user } = req // Logged in user, which has been added to the request object by the middleware form JWT

  if (user.id !== checkIfPostExists.userId) {
    return sendDataResponse(res, 403, {
      message: 'You are only allowed to update your own posts'
    })
  }

I can do this. Seems to be working after testing.
```

Comment (sebgro98) [reply src/controllers/post.js:60]: Can you push this code to the new branch so i can test it

**Review by sebgro98 — DISMISSED** [review]
Everything seems to be working. But i will be waiting to merge Olivers code first as he is doing quite a lot of changes to make sure this one works after

File: src/domain/post.js
Comment (sebgro98) [inline src/domain/post.js:1]: With the new user fields in main the githubUrl was changed to githubUsername can you update that and I will try the code again

**Review by sebgro98 — APPROVED** [review]
Approved

**Comment by amos1969** [comment]
Approved

**Pull request merged**